src/main/java/lotto/domain/Lotto.java

```diff
@@ -0,0 +1,24 @@
+package lotto.domain;
+
+import java.util.List;
+
+public class Lotto {
+    public static final int PRICE = 1_000;
+    public static final int LOTTO_SIZE = 6;
+
+    public List<Integer> numbers;
+
+    public Lotto(List<Integer> numbers) {
+        this.numbers = numbers;
+    }
+
+    public boolean contains(Integer number) {
+        return numbers.contains(number);
+    }
+
+    public int matchCountWith(Lotto other) {
+        return (int) numbers.stream()
+                .filter(number -> other.contains(number))
+                .count();
+    }
```

Comment:
메서드 레퍼런스 사용을 통해 조금 더 간결하게 나타내볼 수도 있겠네요! 😄 
```suggestion
    public int matchCountWith(Lotto other) {
        return (int) numbers.stream()
                .filter(other::contains)
                .count();
    }
```